### PR TITLE
Add redirectUrls prefix to merchant config and redirectUrl to create …

### PR DESCRIPTION
**Pull request by Neeraj-Ghodla**

### CODE REVIEW

File: src/api/payment-requests/payment-requests.md [hunk 1]
```diff
@@ -171,6 +173,14 @@ corresponds to the GPC brick attributes.
       value: { amount: '8991', currency: 'NZD' },
     })
   }
+  example {
+    title 'Create a Payment Request with redirect url'
+    body ({
+      configId: '5efbe2fb96c08357bb2b9242',
+      value: { amount: '8991', currency: 'NZD' },
+      redirectUrl: 'https://app.centrapay.com',
```
Comment (ncjones) [inline src/api/payment-requests/payment-requests.md:181]: can you use a realistic shopping url like https://example.com/store/checkout?cartId=1234

File: src/api/payment-requests/payment-requests.md [hunk 2]
```diff
@@ -61,14 +61,16 @@ version (documented on this page) and the "legacy" version (documented at
...
+| barcode          | String | Scanned patron barcode used to create the payment request.                                                                                                            |
+| expirySeconds    | Number | The expiry seconds used to configure the payment request expiry.                                                                                                      |
+| lineItems        | Array  | **EXPERIMENTAL** The [Line Items](#line-item) being paid for.                                                                                                         |
+| purchaseOrderRef | String | A reference to a purchase order for this payment request.                                                                                                             |
+| invoiceRef       | String | A reference to an invoice for this payment request.                                                                                                                   |
+| redirectUrl      | String | **Experimental** Url to redirect to after paying a payment request. The redirectUrl prefix must be from one of the supported redirectUrls for the [Merchant Config][] |
```
Comment (ncjones) [inline src/api/payment-requests/payment-requests.md:72]: after paying OR cancelling

File: src/api/merchants/merchant-configs.md [hunk 3]
```diff
@@ -24,11 +24,18 @@ A Merchant Config defines the available payment options for paying a [Payment Re
...
+| Field          | Type   | Description                          |
+| :------------  | :----- | :----------------------------------- |
 | paymentOptions | Array  | Array of [Payment Option Configs][]. |
 
 
+{% h4 Optional Fields %}
+
+|    Field     | Type  |                                               Description                                               |
+| :----------- | :---- | :------------------------------------------------------------------------------------------------------ |
+| redirectUrls | Array | **Experimental** List of supported url prefixes that the user can go to after paying a payment request. |
```
Comment (ncjones) [inline src/api/merchants/merchant-configs.md:36]: can we call this "allowedRedirectUrls"?

File: src/api/merchants/merchant-configs.md [hunk 4]
```diff
@@ -24,11 +24,18 @@ A Merchant Config defines the available payment options for paying a [Payment Re
...
+| Field          | Type   | Description                          |
+| :------------  | :----- | :----------------------------------- |
 | paymentOptions | Array  | Array of [Payment Option Configs][]. |
 
 
+{% h4 Optional Fields %}
+
+|    Field            | Type  |                                               Description                                                            |
+| :------------------ | :---- | :------------------------------------------------------------------------------------------------------------------- |
+| allowedRedirectUrls | Array | **Experimental** List of allowewd url prefixes that the user can go to after paying or cancelling a payment request. |
```
Comment (heymatthew) [inline src/api/merchants/merchant-configs.md:36]: I'm thinking this is not going to scale well. A URL is generally inclusive of a resource.

i.e. redirects might go back to something dynamic like https://app.cp42.click/pay/3923067a-5e4e-421b-952b-3e915e2b43d3, and specifying all URLs is a bit excessive.

Are we allowing for more general redirects like https://app.cp42.click/hello/*?
If so, can we rename this to something more general like allowedRedirectFQDNs, or allowedRedirectHosts?

Comment (Ray2126) [reply src/api/merchants/merchant-configs.md:36]: I like allowedRedirectHosts

Comment (Ray2126) [reply src/api/merchants/merchant-configs.md:36]: There's a small type allowewd --> allowed

File: src/api/merchants/merchant-configs.md [hunk 5]
```diff
@@ -24,11 +24,18 @@ A Merchant Config defines the available payment options for paying a [Payment Re
...
+| Field          | Type   | Description                          |
+| :------------  | :----- | :----------------------------------- |
 | paymentOptions | Array  | Array of [Payment Option Configs][]. |
 
 
+{% h4 Optional Fields %}
+
+|    Field            | Type  |                                               Description                                                           |
+| :------------------ | :---- | :------------------------------------------------------------------------------------------------------------------ |
+| allowedRedirectUrls | Array | **Experimental** List of allowed url prefixes that the user can go to after paying or cancelling a payment request. |
```
Comment (ncjones) [inline src/api/merchants/merchant-configs.md:36]:
```suggestion
| allowedRedirectUrls | Array | **Experimental** Allowed prefixes for the `redirectUrl` property on Payment Requests created with this Config. |
```

File: src/api/payment-requests/payment-requests.md [hunk 6]
```diff
@@ -61,14 +61,16 @@ version (documented on this page) and the "legacy" version (documented at
...
+| expirySeconds    | Number | The expiry seconds used to configure the payment request expiry.                                                                                                              |
+| lineItems        | Array  | **EXPERIMENTAL** The [Line Items](#line-item) being paid for.                                                                                                                 |
+| purchaseOrderRef | String | A reference to a purchase order for this payment request.                                                                                                                     |
+| invoiceRef       | String | A reference to an invoice for this payment request.                                                                                                                           |
+| redirectUrl      | String | **Experimental** Url to redirect to after paying or cancelling a payment request. The redirectUrl prefix must be from one of allowedRedirectUrls for the [Merchant Config][]. |
```
Comment (ncjones) [inline src/api/payment-requests/payment-requests.md:72]:
```suggestion
| redirectUrl      | String | **Experimental** URL to redirect the user to after they pay or cancel the Payment Request. Must start with one of the `allowedRedirectUrls` for the [Merchant Config][]. |
```

**Review by ncjones — APPROVED** [review]

File: src/api/merchants/merchant-configs.md (hunk 4 above)
Comment (ncjones) [reply src/api/merchants/merchant-configs.md]: @heymatthew It is a prefix. I thought "allowedRedirectUrlPrefixes" was too wordy so settled on "allowedRedirectUrls". If you read the description it tells you that it is a prefix.

**Review by heymatthew — APPROVED** [review]
